docs/tutorials/posts/omr-aimo2-inference-recipe.md

```diff
@@ -0,0 +1,316 @@
+In the notebook, you can:
+
+- Run inference on different TensorRT-LLM engines (BF16, FP8, FP8+ReDrafter).
+- Compare performance benchmarks.
+- Explore advanced controls like **terminating after the first N generations complete**.
+- Run inference with tool-calling.
+
+Here’s a sample of the kind of benchmark results you’ll see:
+
+| Metric                        | BF16 | FP8   | FP8+ReDrafter  |
+|-------------------------------|---------------|-------|-------|
+| Total Generation Time (s)     | 144.2 | 64.7  |  30.5  |
+| Average Sample Throughput (Tok/s) | 34.6 | 75.2   |  138.5  |
+
+*(full benchmarks and code available in the notebook)*
+
+-----
+
+## 5\. Optional: Enabling tool-calling and the code execution sandbox
+
+The OpenMath LLM is a powerful model that can use Python as a tool integrated in its reasoning. This means it doesn't just generate text; it can also write and execute Python code in a sandbox to solve problems. In the companion notebook, we provide an example how to launch both the LLM server and its accompanying code execution sandbox.
+
+The interaction works like this:
+
+1.  The LLM generates Python code wrapped in `<tool_call>` and `</tool_call>` tokens.
+2.  The inference engine extracts and sends this code to the sandbox.
+3.  The sandbox executes the code and returns the results.
+4.  The output is fed back to the LLM for continued generation or to finalize its answer.
+
+Here's an example of such an interaction:
+<details markdown><summary><b>Show Example Output</b></summary>
+
+Therefore, b = k - 7 = 21 or 49. So, same result. Therefore, sum is 70.
+
+Alternatively, maybe I can write a small program to check for all bases b > 9, compute 9b + 7 and b + 7, check if the latter divides the former, and collect all such bases. Then sum them. Let's do that to verify.
+
+Here's a Python code to perform the check:
+
+```
+    <tool_call>
+    # Initialize a list to store valid bases
+    valid_bases = []
+
+    # Check bases from 10 upwards
+    for b in range(10, 10000):  # Arbitrary large upper limit
+        num1 = 9 * b + 7
+        num2 = b + 7
+        if num1 % num2 == 0:
+            valid_bases.append(b)
+            print(f"Found base: {b}")
+
+    # Sum the valid bases
+    sum_bases = sum(valid_bases)
+    print(f"Sum: {sum_bases}")
+
+    # If sum is over 1000, take modulo 1000
+    if sum_bases > 1000:
+        result = sum_bases % 1000
+    else:
+        result = sum_bases
+
+    print(f"Final Result: {result}")
+    </tool_call>
+    ```output
+    Found base: 21
+    Found base: 49
+    Sum: 70
+    Final Result: 70
+    ```
+```
+The code confirms that the valid bases are 21 and 49, summing to 70.
+
+</details>
+
+To turn off tool-calling in the [companion notebook](../notebooks/demo_aimo_inference.ipynb) use `get_model` instead of `get_code_execution_model` as shown in the NeMo-Skills [docs](https://nvidia.github.io/NeMo-Skills/).
```

Comment:
```suggestion
To turn off tool-calling in the [companion notebook](https://github.com/NVIDIA/NeMo-Skills/tree/main/docs/tutorials/notebooks/demo_aimo_inference.ipynb) use `get_model` instead of `get_code_execution_model` as shown in the NeMo-Skills [docs](https://nvidia.github.io/NeMo-Skills/).
```